### PR TITLE
Bugfixes for GSAC contest

### PR DESCRIPTION
**Pull request by akuleshov7**
### What's done:
- Migrating to FC/VFC of creation components to have proper redirection
- Trying to fix execution details for tests (WIP)

### CODE REVIEW

**Comment by akuleshov7** [comment]
Haven't fixed yet a details on execution view

**Review by kgevorkyan — APPROVED** [review]
could you please add some screenshots?

**Comment by kgevorkyan** [comment]
https://github.com/saveourtool/save-cloud/issues/2613 related?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/modal/LogoutModal.kt [hunk 1]
```diff
@@ -29,7 +32,7 @@ fun logoutModal(
         )
         if (replyToLogout.ok) {
             // logout went good, need either to reload page or to setUserInfo(null) and use redirection like `window.location.href = window.location.origin`
-            window.location.href = "${window.location.origin}/"
+            navigate("/")
             window.location.reload()
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/modal/LogoutModal.kt:36]: maybe set `setUserInfo(null)` ?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/TestStatusComponent.kt [hunk 2]
```diff
@@ -10,47 +12,66 @@ import com.saveourtool.save.core.result.TestStatus
...
  */
-@Suppress("TOO_LONG_FUNCTION")
-fun <D : Any> testStatusComponent(testResultDebugInfo: TestResultDebugInfo, tableInstance: Table<D>) = FC<Props> {
+@Suppress("TOO_LONG_FUNCTION", "LongMethod")
+fun testStatusComponent(
+    organizationProjectPath: String,
+    testResultDebugInfo: TestResultDebugInfo,
+    tableInstance: Table<TestExecutionExtDto>,
+    testExecutionDto: TestExecutionDto,
+) = FC<Props> {
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/TestStatusComponent.kt:42]: `FC<Props>` can be replaced by `VFC`

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt [hunk 3]
```diff
@@ -90,7 +92,8 @@ val registrationView: FC<RegistrationProps> = FC { props ->
             responseHandler = ::responseHandlerWithValidation,
         )
         if (response.ok) {
-            window.location.href = "${window.location.origin}/"
+            props.userInfoSetter(userInfo)
+            useNavigate(to = "/${FrontendRoutes.THANKS_FOR_REGISTRATION}")
             window.location.reload()
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt:97]: do we need it taking into account that userInfoSetter is called?

**Review by nulls — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/TestStatusComponent.kt [hunk 4]
```diff
@@ -10,47 +12,66 @@ import com.saveourtool.save.core.result.TestStatus
...
+    }
+    tr {
+        className = ClassName("table-sm")
+        td {
+            colSpan = 3
             +"Executed command"
         }
         td {
-            colSpan = numColumns - 2
+            colSpan = numColumns - 3
```
[the remark this answers is not in the document]
Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/TestStatusComponent.kt:74]: fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt (hunk 3 above)
Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt:97]: Just for beautiful loading animation so we will have a feeling that user really pressed a button

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/TestStatusComponent.kt (hunk 2 above)
Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/TestStatusComponent.kt:42]: oh, missed that,  thanks

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/modal/LogoutModal.kt (hunk 1 above)
Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/modal/LogoutModal.kt:36]: Do not want to put UserInfo also to topbar... Let it be only in two places, where we modify user

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ExecutionView.kt [hunk 5]
```diff
@@ -309,7 +313,8 @@ class ExecutionView : AbstractView<ExecutionProps, ExecutionState>(Style.SAVE_LI
     } ?: ""
 
     private suspend fun getAdditionalInfoFor(testExecution: TestExecutionDto, id: String) {
-        val trDebugInfoResponse = getDebugInfoFor(testExecution)
+        val trDebugInfoResponse = getDebugInfoFor(testExecution.requiredId())
+        // FixMe: invalid setup of execution because of the invalid propagated ID
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ExecutionView.kt:317]: @nulls this does not work. Looks like it is the one of the reasons why we don't see logs for INTERNAL_ERROR 

I fixed debug info, but this part with execution info was lost while we were migrating to S3.

**Pull request merged**